File: assertion-roulette/python-unittest/GlobalQueryFiltersTests_Should_Add_Deletion_Auditing_Properties.py
import unittest

class TestGlobalQueryFilters(unittest.TestCase):
    def test_should_add_deletion_auditing_properties(self):
        result = default_test_db_context.products.add(Product(name="Product Name", code="product_code"))
        default_test_db_context.save_changes()
        
        deleted_entity = default_test_db_context.products.remove(result.entity)
        default_test_db_context.save_changes()
        
        self.assertIsNotNone(deleted_entity.entity.deleter_user_id)
        self.assertNotEqual("", deleted_entity.entity.deleter_user_id)
        self.assertIsNotNone(deleted_entity.entity.deletion_time)
        self.assertEqual(datetime.today().strftime("%Y-%m-%d"), deleted_entity.entity.deletion_time.strftime("%Y-%m-%d"))

File: assertion-roulette/csharp-xunit/GlobalQueryFiltersTests_Should_Add_Deletion_Auditing_Properties.cs
using Xunit;

namespace BackOfficeBase.Tests.DataAccess
{
    public class GlobalQueryFiltersTests
    {
        [Fact]
        public void Should_Add_Deletion_Auditing_Properties()
        {
            {
                var result = await DefaultTestDbContext.Products.AddAsync(new Product { Name = "Product Name", Code = "product_code" });
                await DefaultTestDbContext.SaveChangesAsync();
                var deletedEntity = DefaultTestDbContext.Products.Remove(result.Entity);
                await DefaultTestDbContext.SaveChangesAsync();
                Assert.NotNull(deletedEntity.Entity.DeleterUserId);
                Assert.NotEqual(Guid.Empty, deletedEntity.Entity.DeleterUserId);
                Assert.NotNull(deletedEntity.Entity.DeletionTime);
                Assert.Equal(DateTime.Today.ToShortDateString(), deletedEntity.Entity.DeletionTime.Value.ToShortDateString());
            }
        }
    }
}
